### PR TITLE
Added base type to the type definition.

### PR DESCRIPTION
**Pull request by jrchudy**
This PR addresses issue #461.

When a base type is present on the type in the jsonColumn, it is now set on the type.

### CODE REVIEW

**Comment by RFSH** [comment]
Didn't we wanted to change the `formatvalue` function accordingly based on the basetype? Or that is going to be another PR? 
Also we can have very simple test cases [here](https://github.com/informatics-isi-edu/ermrestjs/blob/master/test/specs/common/tests/02.column.js).

**Comment by jrchudy** [comment]
We are just exposing the base type if it exists right now. 

Any changes to start managing this in `formatvalue` will be handled in another PR when that issue comes up. I will check with @hongsudt before merging these changes.

**Comment by RFSH** [comment]
We started writing unit tests after implementing the core.js. The test cases in the file that I gave you are for any features that we added to core.js. Eventually we might need to test all the features of the core.js, But at least we should test whatever we're adding.

**Comment by jrchudy** [comment]
@RFSH I factored out the switch statement so it can be used recursively. I can't test this functionality with unit tests currently without modifying a schema through `psql` to define a new domain type.

File: js/core.js [hunk 1]
```diff
@@ -1669,52 +1669,10 @@
...
-                case 'jsonb':
-                    data = utils.printJSON(data);
-                    break;
-                default: // includes 'text' and 'longtext' cases
-                    data = utils.printText(data, options);
-                    break;
-            }
-            return data.toString();
+
+            return _formatValueByType(this.type, data, options).toString();
```
Comment (RFSH) [inline js/core.js]: Why are you calling the `toString` here? If the function is expected to return string (based on jsdoc its return type is string), it should be inside that function and not here.

**Pull request merged**